mlflow/java/client/src/main/java/org/mlflow/tracking/creds/HostCredsProviderChain.java

```diff
@@ -0,0 +1,47 @@
+package org.mlflow.tracking.creds;
+
+import org.apache.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.mlflow.tracking.MlflowClientException;
+
+public class HostCredsProviderChain implements MlflowHostCredsProvider {
+  private static final Logger logger = Logger.getLogger(HostCredsProviderChain.class);
+
+  private final List<MlflowHostCredsProvider> hostCredsProviders = new ArrayList<>();
+
+  public HostCredsProviderChain(MlflowHostCredsProvider... hostCredsProviders) {
+    this.hostCredsProviders.addAll(Arrays.asList(hostCredsProviders));
+  }
+
+  @Override
+  public MlflowHostCreds getHostCreds() {
+    List<String> exceptionMessages = new ArrayList<>();
+    for (MlflowHostCredsProvider provider : hostCredsProviders) {
+      try {
+        MlflowHostCreds hostCreds = provider.getHostCreds();
+
+        if (hostCreds != null && hostCreds.getHost() != null) {
+          logger.debug("Loading credentials from " + provider.toString());
+          return hostCreds;
+        }
+      } catch (Exception e) {
```

Comment:
Maybe safer to catch Throwable?

Reply:
We don't want to catch things like OutOfMemoryError, as that could cause the JVM to be unstable -- Scala's NonFatal is more suitable, but in Java most authors leave it at Exception.

Reply:
Yeah OOM makes sense but any Error will not be caught as Exception.

Reply:
But would be caught by Throwable